---
title: "MICB425 Portfolio - Microbial Ecological Genomics: From Genomes to Biomes, Spring 2018"
author: "Zijie John Zhu (11656148)"
date: "version `r format(Sys.time(), '%B %d, %Y')`"
output: 
  html_document:
    toc: true
    toc_float:
      collapsed: yes
---

# Module 01

## Module 01 portfolio check
* Installation check
    + Completion status:
    + Comments:
* Portfolio repo setup
    + Completion status: 
    + Comments:
* RMarkdown Pretty PDF Challenge
    + Completion status:
    + Comments:
* Evidence worksheet_01
    + Completion status:
    + Comments:
* Evidence worksheet_02
    + Completion status:
    + Comments:
* Evidence worksheet_03
    + Completion status:
    + Comments:
* Problem Set_01
    + Completion status:
    + Comments:
* Problem Set_02
    + Completion status:
    + Comments:
* Writing assessment_01
    + Completion status:
    + Comments:
* Additional Readings
    + Completion status:
    + Comments  
    
## Data science Friday  

### Installation check
![](module_1/screenshot_git_bash.png)
![](module_1/screenshot_rstudio.png)
![](module_1/screenshot_github.png)

### Portfolio repo setup

*Set up Git user locally*
```
git config --global user.name "johnzzhu"
git config --global user.email $EMAIL
```

*Creating local repository*
```
cd $REPO_DIR
git init
git add .
git commit -m "Initial commit"
git remote add origin $REPO_URL
git remote -v
git push -u origin master
```

### RMarkdown pretty PDF challenge
````
`r paste(readLines('module_1/pretty_html.Rmd', encoding = "UTF-8"), collapse = '\n')`
````

### Plotting data in R

Installing Phyloseq
``` {r eval=FALSE}
source("https://bioconductor.org/biocLite.R")
biocLite("phyloseq")
```

Load libraries & data
``` {r message=FALSE}
library(tidyverse)
library(phyloseq)
metadata=read.table(file="Saanich.metadata.txt", header=TRUE, row.names=1, sep="\t", na.strings="NAN")
load("phyloseq_object.RData")
```

Plot 1: Simple scatterplot
``` {r}
ggplot(metadata, aes(x=PO4_uM, y=Depth_m)) + 
  geom_point(shape=17, color="purple")
```

Plot 2: Plot dplyr mutated data
``` {r}
metadata %>% 
  mutate(Temperature_F = Temperature_C * 9/5 + 32) %>%  # convert C to F
  ggplot(aes(x=Temperature_F, y=Depth_m)) +
  geom_point()
```

Plot 3: Plot using Phyloseq mutated data & adding labels
``` {r}
physeq_percent = transform_sample_counts(physeq, function(x) 100 * x/sum(x)) # transform to relative abundance
plot_bar(physeq_percent, fill="Domain") + 
  geom_bar(aes(fill=Domain), stat="identity") +
  labs(y="Percent Relative Abundance by Domain", x="Sample Depth Below Water Surface (m)", 
       title="Distribution of Domains at Various Sample Depths")
```

Plot 4: Faceted plot using tidied data 
``` {r}
metadata %>% 
  select(Depth_m, NH4_uM, NO2_uM, NO3_uM, O2_uM, PO4_uM, SiO2_uM) %>% 
  gather(Nutrient,uM,-Depth_m) %>% # tidy the data
  ggplot(aes(x=Depth_m, y=uM)) + 
  geom_line() + geom_point() + # line plot with points
  facet_wrap(~Nutrient, scales="free_y") + # facet the plot
  theme(legend.position="none") # remove the legend
```


## Origins and Earth Systems
### Evidence worksheet 01
[Whitman *et al* 1998](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)

#### Learning objectives
Describe the numerical abundance of microbial life in relation to ecology and biogeochemistry of Earth systems. 

#### General questions
* What were the main questions being asked?

    What is the total number, density, carbon mass, growth and turnover rates of prokaryotic organisms in aquatic, soil, and subsurface environments?
    
* What were the primary methodological approaches used?
    
    This study is a review of other published literature. The authors presented data from other studies of aquatic, soil, and subsurface environments, and presented them in a tabular form. The authors then extrapolate the remaining data so that values of cell densities, abundances, and carbon contents are complete for each environment. Lastly the authors conduct comparative analyses based on these data and address these questions.  

* Summarize the main results or findings.  

    * Total prokaryotic carbon on Earth is estimated to be 60~100% of the carbon mass found in plants.
    * Prokaryotes have approx. one order of magnitude more of total N and P mass than that in plants, because they contain much more N and P relative to carbon than plants do.
    * Total protoplasmic biomass on Earth in microbes is far higher than plants
    * Lower limit in prokaryotic turnover time ranges from 6-25 days to ~1000 days (1-2 thousand years for subsurface bacteria, however it is far longer than expected and is based on one sample) 
    * 0.4, 0.5, 3.4, and 170 hours avg for 4 mutations from the pool of all genes shared by marine heterotrophs (in the upper 200 m), marine autotrophs, soil prokaryotes, or prokaryotes in domestic animals
    * 60 years avg for 5 mutations from the pool of all genes shared by all 4 groups    


* Do new questions arise from the results?  

    Why is the subsurface turnover rate so slow?  
    Which hypothesis is true: metabolic inactivity, lithotrophic metabolism, overestimation of subsurface biomass, underestimation of carbon assimilation?  

* Were there any specific challenges or advantages in understanding the paper (*e.g.* did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)?  

    Could clarify what is “four simultaneous mutations in every gene shared by the populations of marine heterotrophs” a bit more. 


### Problem set 01
#### Learning objectives
Describe the numerical abundance of microbial life in relation to the ecology and biogeochemistry of Earth systems.

#### Specific questions
* What are the primary prokaryotic habitats on Earth and how do they vary with respect to their capacity to support life? Provide a breakdown of total cell abundance for each primary habitat from the tables provided in the text.  

    The habitats are **aquatic, soil,** and **subsurface**  
    
    Total cell abundances:  
    
    Aquatic: \(1.181E29\) cells  
    Soil:   \(2.556E29\) cells  
    Subsurface: \(3.8E30\) cells  
    
    ![](module_1/whitman_table1.PNG)  
    ![](module_1/whitman_table2.PNG)  
    ![](module_1/whitman_table3.PNG)  

* What is the estimated prokaryotic cell abundance in the upper 200 m of the ocean and what fraction of this biomass is represented by marine cyanobacterium including Prochlorococcus? What is the significance of this ratio with respect to carbon cycling in the ocean and the atmospheric composition of the Earth?  

    The estimated prokaryotic cell abundance is 5E5 cells/mL, or 360E26 cells total.  
    Marine cyanobacteria & Procholorococcus account for 4E4 of the cells per mL, or 8% of all prokaryotic cells in the top 200m.  
    They are responsible for atmospheric oxygenation and carbon assimilation.

* What is the difference between an autotroph, heterotroph, and a lithotroph based on information provided in the text?   

    Autotroph: Energy source independent of organic substrates  
    Heterotroph: Energy source from organic carbon substrates  
    Lithotroph: Energy source from inorganic substrates  

* Based on information provided in the text and your knowledge of geography what is the deepest habitat capable of supporting prokaryotic life? What is the primary limiting factor at this depth?  
    
    At a subsurface depth of 4km, the average temperature is 125C.  
    This is close to the temperature limit for prokaryotic life.  
    The high temperature is likely due to geothermal processes.  
    On average, the temperature increases by 22 degrees Celsius per km depth.

* Based on information provided in the text your knowledge of geography what is the highest habitat capable of supporting prokaryotic life? What is the primary limiting factor at this height?  

    Prokaryotic life has been detected in the air at very low qualtities up to 57-77km (more likely 20km realistically).  
    The low temperature & pressure as well as extreme solar radiation are limiting factors at this height.  
    However it is arguable that the atmosphere is capable of supporting growth (as opposed to transport) due to the lack of many essential nutrients at any appreciable concentrations.

* Based on estimates of prokaryotic habitat limitation, what is the vertical distance of the Earth’s biosphere measured in km?  

    If including the atmosphere, Earth's biosphere spans from approx. 20km above the summit of Mt. Everest to approx. 4.5km below the Marianas Trench. This amounts to approx. 44km. 

* How was annual cellular production of prokaryotes described in Table 7 column four determined? (Provide an example of the calculation)  

    Annual cellular production is the population size mutiplied by the number of turnover periods in a year.  
    For marine heterotrophs above 200m:  
    Population size = \(3.6E28\) cells  
    Turnover period = \(16\) days  
    Annual cellular production = population * turnover/yr = \(3.6E28\frac{365}{16} = 8.215E29\) cells/yr  
    *Aside*: Viruses play a significant role in driving this turnover. Viruses often carry accesorry metabolism genes (AMG) which directly modulates cellular metabolism. At a larger scale, this in turn modulates the metabolism of the community.

* What is the relationship between carbon content, carbon assimilation efficiency and turnover rates in the upper 200m of the ocean? Why does this vary with depth in the ocean and between terrestrial and marine habitats?  

    A number of assumptions need to be made to establish these relationships.  
    
    Assuming:  
    20fg = 20E-30Pg of C per cell and  
    3.6E28 marine heterophs in the upper 200m  
    We can estimate that there are:  
    3.6E28 * 2E-30 = 0.72Pg C in marine heterotrophs in the upper 200m  
    
    Assuming:  
    20% carbon assimilation efficiency (growth efficiency, substrate use efficiency)  
    We can estimate that amount of C required per turnover to be:  
    4 * 0.72 = 2.88Pg (unclear why the author used a factor of 4 instead of 5)  
    
    Assuming:  
    The net primary production in terms of C is 51Pg/yr and  
    85% of it is consumed in the top 200m, all by prokaryotes  
    We can estimate that the amount of C available per year is:  
    51 * 0.85 = 43Pg/yr  
    
    This works out to 43/2.88 = 14.9 turnovers per year or 24.5 days  

* How were the frequency numbers for four simultaneous mutations in shared genes determined for marine heterotrophs and marine autotrophs given an average mutation rate of 4 x 10^-7^ per DNA replication? (Provide an example of the calculation with units. Hint: cell and generation cancel out)  

    The probability of 4 simultaneous mutations would be (4E-7)^4^ = 2.56E-26 per replication.  
    
    Given the turnover rate to be 8.215E29 cells/yr from a previous question, then there is an average of 8.2E29 * 2.56E-26 = 2.1E4 four simultaneous mutation events per year, or about 0.4 per hour.


* Given the large population size and high mutation rate of prokaryotic cells, what are the implications with respect to genetic diversity and adaptive potential? Are point mutations the only way in which microbial genomes diversify and adapt?  

    The prokaryotic genomes have tremendous abilities to change given high mutation rates, large population size, and rapid turnover rates. By accumulating mutations not only through point mutations but also through insertion, deletion, HGT and recombination, mutations can rapidly spread horizontally and vertically. This makes the world of prokaryotes incredibly diverse and adaptable to any kind of biologically viable habitat.

* What relationships can be inferred between prokaryotic abundance, diversity, and metabolic potential based on the information provided in the text?  
    
    Since there is high diversity, there is also a wide range of metabolic capabilities.  
    Since there is also high abundance, the impact/extent of metabolism on the Earth is also very large - enough to oxygenate the atmosphere, for example.


### Evidence Worksheet 02 “Life and the Evolution of Earth’s Atmosphere”
#### Learning objectives
Comment on the emergence of microbial life and the evolution of Earth systems
    
    
* Indicate the key events in the evolution of Earth systems at each approximate moment in the time series. If times need to be adjusted or added to the timeline to fully account for the development of Earth systems, please do so. 

    + 4.6 billion years ago  
      Inner solar system formed  
      Water vapour & carbon (as CO2 in limestone) deposited on Earth  
      Hot (500C)

    + 4.5 billion years ago  
      Moon forming impact led to Earth spin, tilt, generated diurnal cycle & seasons
      Rock vapour atmosphere formed (optically thick)
      Hot (~100C) for 2 million years
      Late heavy bombardments - Frequent large scale bombardments raise temp over 100C
      
    + 4.4 billion years ago
      Zircons formed
      
    + 4.1 billion years ago
      Evidence of life found in graphite in Zircons

    + 4.0 billion years ago  
      Evidence of plate subduction
      Oldest rock - Acasta gneiss from Canada

    + 3.8 billion years ago  
      Bombardments halted  
      Oceans formed  
      Evidence of sedimentary rocks  
      Seawater chemistry stabilized
      Evidence of life in carbon isotopes too
    
    + 3.75 billion years ago  
      Evidence of photosynthesis  
    
    + 3.5 billion years ago  
      Microfossils, stromatolites 
      Evidence of isotopic partition of C from carbonate
      Most likely Rubisco global oxygenic photosynthesis 
    
    + 3.0 billion years ago  
      First glaciation event - oxygen stops methanogensis
    
    + 2.7 billion years ago  
      Evidence of cyanobacteria & eukaryotes
      
    + 2.5 billion years ago  
      second glaciation event???

    + 2.2 billion years ago  
      Great oxidation event - sharp increase in O2 (may involve complex eukaryotes)  
      Evidence of redbeds containing iron oxide  
      Regulation of mitochondria & chloroplasts may modulate balance of photosynthesis, CO2 & N fixation
    
    + 2.1 billion years ago
      Oxygenic photosynthesis in cyanobacteria??????
      
    + 1.9 billion years ago
      Eukayotes
      
    + 1.2 billion years ago
      Evidence of sexual reproduction
      
    + 500 million years ago
      Cambrian explosion
      
    + 400 million years ago
      Devonian explosion - land plants
      Fish, cephalopods, corals formed
      
    + 300 million years ago
      Gigantism
      Formation of Pangae
 
    + 250 million years ago
      Permian extinction
      
    + 240 million years ago
      Evidence of dinosours
      
    + 200 million years ago
      Triassic Jurassic extinction event
      
    + 60 million years ago
      Creataceous Tertiary extinction event
    
    + 550,000 years ago
      Archaic humans first appear
    
    + 200,000 years ago
      Homo sapiens first appear
      
Early Archean
S processing archaea (hyperthermophile)
Methanogens (hyperthermophile)
Lower temp methanogens appeared later

* Describe the dominant physical and chemical characteristics of Earth systems at the following waypoints:  

    + Hadean  
    Hot, molten
    
    + Archean  
    Early sun dim, 30% dimmer in archarern <, would freesze without greenhouse
    Methanogenesis kept Earth from freezing
    
    + Precambrian  
    
    + Proterozoic  
    
    increased complexity & oxygen in proterozoic
    
    + Phanerozoic  
    

### Problem set 02 “Microbial Engines”
#### Learning objectives
Discuss the role of microbial diversity and formation of coupled metabolism in driving global biogeochemical cycles.

#### Specific Questions
* What are the primary geophysical and biogeochemical processes that create and sustain conditions for life on Earth? How do abiotic versus biotic processes vary with respect to matter and energy transformation and how are they interconnected?  

    The primary geophysical and biogeochemical processes are plate tectonics & atmospheric photochemical processes. They allow distribution, resupply, and reaction between chemical species & nutrients.

    Abiotic processes are primarily acid-base (H ions), because most of molecular H has escaped to space during Earth's formation, whereas biotic processes are primarily redox (electrons).

* Why is Earth’s redox state considered an emergent property?  

    Since biological processes are largely redox based, and Earth's abiotic processes interacts and ultimately drive the biological processes, Earth's redox state is a property tightly linked to its biosphere, thus it's considered an emergent property.  

* How do reversible electron transfer reactions give rise to element and nutrient cycles at different ecological scales? What strategies do microbes use to overcome thermodynamic barriers to reversible electron flow?  


* Using information provided in the text, describe how the nitrogen cycle partitions between different redox “niches” and microbial groups. Is there a relationship between the nitrogen cycle and climate change?   

    N cycling involves many different redox niches and microbial group.
    Nitrogen fixation is carried out solely anaerobically by nitrogenase.
    Ammonia is oxidized to nitriate or nitrite aerobically by nitrifying bacteria.
    Another reductive process used by yet other microbes anaerobically reduce ntrite and nitrate to dinitrogen.
    Different microbes that run parts of this cycle are often segregated in time and/or space, due to different requirements in reaction conditions.

    N cycling and climate change may be related in that nitrifying bacteria undergo carbon fixation & assimilation by reducing CO2 while oxidizing ammonia, thus reducing atmospheric CO2.

* What is the relationship between microbial diversity and metabolic diversity and how does this relate to the discovery of new protein families from microbial community genomes?  

    Metabolic and microbial diversity are often, but not always correlated. For example, the presence of fundamental metabilic processes does not reflect the actual microbial diversity. Neither do simple pathways such as rhodopsin based photosynthesis.  

    New protein families - may exist in other microbes too???

* On what basis do the authors consider microbes the guardians of metabolism?  

    Since the "core planetary gene set" which enables the fundamental metabilic processes is widespread in diverse species of microbes, the extinction of even the majority of the species does not necessarily mean the loss of this set. This microbes as a whole are considered as guardians of metabolism.

### Additional questions???

Are the niches for all possible redox rxns occupied by microbial metabolism?  

    No, there are some niches that most likely cannot be occupied by microbial metabolism.
    All chemical reactions need to overcome activation energies, regardless of the differences in free energies. This implies that reactions with high activation energies may not sometimes be feasible biologically, especially in a highly competitive environment because it may be difficult to capture enough energy for the reaction to proceed.
    This is reflected in the lack of eviddence for the direct oxidation of dinitrogen to nitrate, or the use of ammonia as electron donors for photosynthesis.
    A similar issue exists involving solid electron donors/acceptors, where the organism has to invest energy to transport electrons to/from the substrate.
    

apparent diversity
    Each microbial genome contains a core planetary gene set, along with a set of accessory/boutique genes which may not be common to all microbes.
    Core planetry gene set contains genes relating to (DNA replication, transcription, translation, stress proteins, etc.). These are conserved functionally to maintain a set of core functionns common to survival, and this often leads to genes conserved phylogeneticallly.
    The accessory genes allow for specialized functions.


how does the ancietn
    HGT to diff organisms
    Pathways tend to be resilient because parts or all of a pathway are usually distributed over multiple species as the result of HGT.
    Even if a species does not phenotypically present a function, it does not necessarily mean that such genes do not exist in its genome, due to the regulation of gene expression. Genotype at the individual level does not necessaruly equate with that at the population level either, by the same argument. 
    It can be argued that pathways are sometimes emergent, as it is not observed in a isolated species, but rather observed in a population.
    
    
do machine metaphors
    Similar:
        Microbes both have inputs and outputs
        Machines evolve over time by improvments in design, and microbes accumualate mutations
        Both consume chemical energy 
    Different:
        We understanding machines but we don't fully understanding microbes
        
        
### Module 01 references
Whitman WB, Coleman DC, and Wiebe WJ. 1998. Prokaryotes: The unseen majority. Proc Natl Acad Sci USA. 95(12):6578–6583.  [PMC33863](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)